extensions/core/runtime/src/main/java/org/apache/camel/quarkus/core/runtime/CamelRecorder.java

```diff
@@ -16,88 +16,98 @@
-                try {
-                    runtime.getValue().stop();
-                } catch (Exception e) {
-                    throw new RuntimeException(e);
+        try {
+            if (buildTimeConfig.disableJaxb) {
+                context.adapt(ExtendedCamelContext.class).setModelJAXBContextFactory(() -> {
+                    throw new UnsupportedOperationException();
+                });
+            } else {
+                // The creation of the JAXB context is very time consuming, so always prepare it
+                // when running in native mode, but lazy create it in java mode so that we don't
+                // waste time if using java routes
+                if (ImageInfo.inImageBuildtimeCode()) {
+                    context.adapt(ExtendedCamelContext.class).getModelJAXBContextFactory().newJAXBContext();
                 }
             }
-        });
-    }
+        } catch (Exception e) {
+            throw RuntimeCamelException.wrapRuntimeCamelException(e);
+        }
 
-    public void addBuilder(
-        RuntimeValue<CamelRuntime> runtime,
-        String className) {
+        FastModel model = new FastModel(context);
 
-        FastCamelRuntime fcr = (FastCamelRuntime) runtime.getValue();
+        context.setModel(model);
+        context.init();
 
-        try {
-            fcr.getBuilders().add((RoutesBuilder) Class.forName(className).newInstance());
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
+        // register to the container
+        beanContainer.instance(CamelProducers.class).setContext(context);
+
+        return new RuntimeValue<>(context);
     }
 
     public void bind(
-        RuntimeValue<CamelRuntime> runtime,
+        RuntimeValue<Registry> runtime,
         String name,
         Class<?> type,
         Object instance) {
 
-        runtime.getValue().getRegistry().bind(name, type, instance);
+        runtime.getValue().bind(name, type, instance);
     }
 
     public void bind(
-        RuntimeValue<CamelRuntime> runtime,
+        RuntimeValue<Registry> runtime,
         String name,
         Class<?> type) {
 
         try {
-            runtime.getValue().getRegistry().bind(name, type, type.newInstance());
+            runtime.getValue().bind(name, type, type.newInstance());
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
-    public BeanContainerListener initRuntimeInjection(RuntimeValue<CamelRuntime> runtime) {
-        return container -> container.instance(CamelProducers.class).setCamelRuntime(runtime.getValue());
+    public Supplier<CamelConfig.BuildTime> buildTimeSupplier(CamelConfig.BuildTime config) {
+        return new Supplier<CamelConfig.BuildTime>() {
+            @Override
+            public CamelConfig.BuildTime get() {
+                return config;
+            }
+        };
+    }
+
+    public Supplier<CamelConfig.Runtime> runtimeSupplier(CamelConfig.Runtime config) {
```

Comment:
I vote for renaming to `runtimeConfigSupplier`

Reply:
not used any more, will remove it